libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -636,6 +637,9 @@ bool AP_Mission::read_cmd_from_storage(uint16_t index, Mission_Command& cmd) con
         return false;
     }
 
+    // ensure all bytes of cmd are zeroed
+    cmd = {};
```

Comment:
Why don't we do this once before the if? (Ideally maybe even before taking the semaphore). It looks like code duplication.

Reply:
So as not to modify the passed-in parameter on failure.  Might save a caller a bug at some stage.

Reply:
Yeah, realized I hadn't quite expanded enough lines inbetween the two changes, sorry.